### PR TITLE
Generalized Gaussian blur

### PR DESCRIPTION
**Pull request by joeatodd**
I've replaced the previously hardcoded 9-tap optimized blur with
functions which enable defining arbitrary Gaussian kernels with a sigma value & blur
window. The offsets & weights for this are calculated once during
setup, then sent to static array uniforms for the shader.

By setting a high sigma (10.0) & wide window (49), we can achieve good
blurring in a single iteration. In other words, the 'ping-pong' now
only really ping-pongs once (for each blur direction).

### CODE REVIEW

File: src/renderer_gl.cpp [hunk 1]
```diff
@@ -259,14 +271,15 @@ void RendererGL::render(glm::mat4 proj_mat, glm::mat4 view_mat) {
    glUseProgram(programBlur.getId());
 
    // Blur pingpong (N horizontal blurs then N vertical blurs)
-   if (0) {
+
+   if (1) {
```
Comment (Ruyk) [inline src/renderer_gl.cpp]: if 1 ? why is this needed?

File: src/renderer_gl.cpp [hunk 2]
```diff
@@ -259,14 +271,15 @@ void RendererGL::render(glm::mat4 proj_mat, glm::mat4 view_mat) {
...
       int loop = 0;
       for (int i = 0; i < 2; ++i) {
          if (i == 0)
             glProgramUniform2f(programBlur.getId(), 1, 1, 0);
          else
             glProgramUniform2f(programBlur.getId(), 1, 0, 1);
-         for (int j = 0; j < 100; ++j) {
+         for (int j = 0; j < 1; ++j) {
```
Comment (Ruyk) [inline src/renderer_gl.cpp]: why a for up to 1? either remove the for loop or alternatively put the 1 on a variable with a name that explains why is just one

**Review by Ruyk — COMMENTED** [review]
Some minor comments

File: src/renderer_gl.cpp (hunk 2 above)
Comment (joeatodd) [reply src/renderer_gl.cpp]: Yeah good point, done.
